c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -373,6 +373,29 @@ private:
   kj::HttpService::ConnectResponse& connResponse;
 };
 
+class HttpOverCapnpFactory::TlsStarterImpl final
+    : public TlsStarter::Server {
+public:
+  TlsStarterImpl(kj::TlsStarterCallback tlsStarter)
+      : tlsStarter(kj::mv(tlsStarter)) {}
+
+  kj::Promise<void> startTls(StartTlsContext context) override {
+    KJ_REQUIRE(!tlsStarted, "already called startTls()");
```

Comment:
We may also need to prevent this from being called if the stream has already been destroyed. I think this could be accomplished by having the `EofDetector` hold on to a reference to the callback and set a flag on it when done that revokes it.